src/connectivity.js

```diff
@@ -137,17 +155,42 @@ var ConnectivityChecker = {
         return reqOptions;
       };
 
+
       if (RdGlobalConfig.proxy) {
+        var proxyOpts = url.parse(RdGlobalConfig.proxy.host + ":" +RdGlobalConfig.proxy.port);
+        if (RdGlobalConfig.proxy.username && RdGlobalConfig.proxy.password) {
+          proxyOpts.auth = RdGlobalConfig.proxy.username + ":" + RdGlobalConfig.proxy.password;
+        }
+
         ConnectivityChecker.connectionChecks.push(this.httpToHubWithProxy, this.httpToRailsWithProxy);
         /* eslint-disable-next-line no-unused-vars */
-        ConnectivityChecker.reqOpsWithProxy = function (reqUrl, reqType) {
+        ConnectivityChecker.reqOpsHttpWithProxy = function (reqUrl, reqType) {
+          var parsedUrl = url.parse(reqUrl);
+          var reqOptions = {
+            method: 'GET',
+            headers: {},
+            host: parsedUrl.hostname,
+            port: parsedUrl.port || 80,
+            path: parsedUrl.path,
+            agent: new HttpProxyAgent(proxyOpts)
+          };
+          if (RdGlobalConfig.proxy.username && RdGlobalConfig.proxy.password) {
+            reqOptions.headers['Proxy-Authorization'] = Utils.proxyAuthToBase64(RdGlobalConfig.proxy);
+          }
```

Comment:
do we need this anymore? Since line 161-163, we are adding the auth to `proxyOpts`.. which we are passing to `agent`... can we verify this once?

Reply:
similarly below...

Reply:
Changed and tested. Working fine.

Reply:
can be resolved